ee/tabby-webserver/src/service/answer.rs

```diff
@@ -490,6 +507,94 @@ Remember, don't blindly repeat the contexts verbatim. When possible, give code s
     )
 }
 
+/// Combine code snippets from search results rather than utilizing multiple hits: Presently, there is only one rule: if the number of lines of code (LoC) is less than 200, and there are multiple hits (number of hits > 1), include the entire file.
+pub async fn merge_code_snippets(
+    repository: Option<Repository>,
+    hits: Vec<CodeSearchHit>,
+) -> Vec<CodeSearchHit> {
+    let repository = match repository {
+        Some(repo) => repo,
+        None => return hits,
+    };
+
+    // group hits by filepath
+    let mut file_hits: HashMap<String, Vec<CodeSearchHit>> = HashMap::new();
+    for hit in hits.clone().into_iter() {
+        let key = format!("{}-{}", repository.source_id, hit.doc.filepath);
+        file_hits.entry(key).or_default().push(hit);
+    }
+
+    let mut result = Vec::with_capacity(file_hits.len());
+
+    for (_, file_hits) in file_hits {
+        if file_hits.len() > 1 {
+            // construct the full path to the file
+            let path: PathBuf = repository.dir.join(&file_hits[0].doc.filepath);
+            let all_lines = match read_file_content(&path) {
+                Ok(lines) => lines,
+                Err(e) => {
+                    warn!("Error reading file {}: {}", path.display(), e);
+                    //cannot read the file, just extend the hits
+                    result.extend(file_hits);
+                    continue;
+                }
+            };
+
+            if !all_lines.is_empty() {
+                let mut insert_hit = file_hits[0].clone();
+                insert_hit.scores =
+                    file_hits
+                        .iter()
+                        .fold(CodeSearchScores::default(), |mut acc, hit| {
+                            acc.bm25 += hit.scores.bm25;
+                            acc.embedding += hit.scores.embedding;
+                            acc.rrf += hit.scores.rrf;
+                            acc
+                        });
+                // average the scores
+                let len = file_hits.len() as f32;
+                insert_hit.scores.bm25 /= len;
+                insert_hit.scores.embedding /= len;
+                insert_hit.scores.rrf /= len;
+                insert_hit.doc.body = all_lines.join("\n");
+                result.push(insert_hit);
+            }
+        } else {
+            result.extend(file_hits);
+        }
+    }
+    result
+}
+
+#[derive(Debug)]
+// Error for when the file is too large
+struct FileTooLargeError;
+
+impl fmt::Display for FileTooLargeError {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "File is too large (more than 200 lines)")
+    }
+}
+
+impl Error for FileTooLargeError {}
+
+//directly read the file content
```

Comment:
```suggestion
/// Read file content, ....
```

Document it will return nothing if line content > 200.